cms/plugin_rendering.py

```diff
@@ -125,12 +125,30 @@ def render_placeholder(placeholder, context_to_copy, name_fallback="Placeholder"
     request = context['request']
     if not hasattr(request, 'placeholders'):
         request.placeholders = {}
-    perms = (placeholder.has_change_permission(request) or not placeholder.cache_placeholder)
-    if not perms or placeholder.slot not in request.placeholders:
-        request.placeholders[placeholder.slot] = (placeholder, perms)
+
+    # Prepend frontedit toolbar output if applicable
```

Comment:
maybe write this as :

```python

try:
	toolbar = getattr(request, 'toolbar', None)
except AttributeError:
	toolbar = None

if (toolbar and toolbar.edit_mode and toolbar.show_toolbar
     and placeholder.is_editable and editable):
     from cms.middleware.toolbar import toolbar_plugin_processor
     processors = (toolbar_plugin_processor, )
     edit = True
 else:
     processors = None
     edit = False
```

`Placeholder` model has `is_editable` as an attribute so no need to use `getattr`